reverseproxy/sni.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2017 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package reverseproxy
+
+import (
+	"io"
+	"net"
+	"net/url"
+
+	"github.com/nu7hatch/gouuid"
+	"github.com/tsuru/planb/log"
+)
+
+type SNIReverseProxy struct {
+	ReverseProxyConfig
+}
+
+func (rp *SNIReverseProxy) Initialize(rpConfig ReverseProxyConfig) error {
+	rp.ReverseProxyConfig = rpConfig
+	return nil
+}
+
+func (rp *SNIReverseProxy) Stop() {
+	// no special treatment for fast reverse proxy
+}
+
+func (rp *SNIReverseProxy) Listen(listener net.Listener) {
+	for {
+		connection, err := listener.Accept()
+		ConnID, _ := uuid.NewV4()
```

Comment:
Could you change `ConnID` to `connID` here and in other references. Capitalized local variables can end up being mistaken for public package variables when reading references to it later.